src/helpers/models.rs

```diff
@@ -0,0 +1,143 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::{Debug, Formatter};
+use std::ops::Range;
+
+// Type aliases to indicate whether the parameter should be encrypted, secret shared, etc.
+// Underlying types are temporalily assigned for PoC.
+type CipherText = [u8; 32];
+type PlainText = String;
+type SecretShare = [CipherText; 3];
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct Event {
+    /// Name of the business entity that generated this event. For source-fanout,
+    /// it's the business who promotes ads. For trigger-fanout, it's the business
+    /// where a user has made a conversion.
+    entity_name: PlainText,
+
+    /// Secret shared and then encrypted match keys.
+    matchkey: SecretShare,
+
+    /// Date and time of the event occurence. Secret shared and encrypted.
+    timestamp: SecretShare,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct SourceEvent {
+    event: Event,
+
+    /// Campaign ID of the ad served to the user.
+    campaign_id: PlainText,
+}
+
+#[cfg(feature = "debug")]
+impl Debug for SourceEvent {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "SourceEvent:\n  matchkey={:?}\n  campaign_id={}",
+            &self.event.matchkey, &self.campaign_id
+        )
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct TriggerEvent {
+    event: Event,
+
+    /// Conversion value.
+    value: SecretShare,
+
+    /// Zero knowledge proom that the trigger value lies within a specific range
+    /// of values. The range is specified in [TriggerFanoutQuery].
+    zkp: PlainText,
+}
+
+#[cfg(feature = "debug")]
+impl Debug for TriggerEvent {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "TriggerEvent:\n  matchkey={:?}\n  value={:?}",
+            &self.event.matchkey, &self.value
+        )
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+enum QueryType {
+    Source,
+    Trigger,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+enum Node {
+    Helper1,
+    Helper2,
+    Helper3,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct Query {
+    /// Caller authentication token.
+    auth_token: PlainText,
+
+    /// Initial MPC helper node to send the data to.
+    leader_node: Node,
+
+    /// List of match key providers to be used by the source and trigger events during an epoch.
+    mk_providers: Vec<String>,
+
+    /// Source-fanout or Trigger-fanout.
+    query_type: QueryType,
+
+    /// Percentage of epoch-level privacy budget this query should consume.
+    privacy_budget: f32,
```

Comment:
We probably cannot support arbitrarily fine-grained queries. I suspect an integer value would be better than a float. For example, an integer ranging from 1 to 100 might actually be better (i.e. we might want to have a cap on the total number of queries can can make).